client/src/contexts/UserStreamContext.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { createContext, useReducer, useCallback, useMemo } from "react";
+
+export type UserStream = MediaStream | null;
+
+interface Action {
+  type: string;
+  id: string;
+  data: UserStream;
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export const UserStreamContext = createContext<any>({});
+
+export const UserStreamProvider = ({ children }: { children: JSX.Element }): JSX.Element => {
+  const reducer = (userStreams: { [key: string]: UserStream }, action: Action) => {
+    switch (action.type) {
+      case "update": {
+        if (!userStreams) {
+          return {};
+        }
+
+        return { ...userStreams, [action.id]: action.data };
+      }
+      case "remove": {
+        // eslint-disable-next-line @typescript-eslint/no-unused-vars
+        const { [action.id]: _toRemove, ...removed } = userStreams;
+        return removed;
+      }
+      default:
+        return userStreams;
+    }
+  };
+
+  const [userStreams, dispatch] = useReducer<React.Reducer<{ [key: string]: UserStream }, Action>>(
+    reducer,
+    {}
```

Comment:
would be friendly to have initialState or defaultState, so that it would be easier to make connection with other parts L12, L19

Reply:
Actually I don't think this block: https://github.com/Amijakan/mochatto/pull/371/files/b6e9f9869d675303160a5954f517993e2767a884#diff-84eea4daa8e5b633288d0e9d01f69756864e1592d12c5faf6018596135bfedffR18-R20
is needed so I'll just get rid of that. I don't think a default state like the UserInfo is needed since this is just a mapping of id's to MediaStreams.